core/field_colour_slider.js

```diff
@@ -0,0 +1,346 @@
+ * @param {Node} node - The DOM node the gradient will be set on.
+ * @param {string} channel – Either "hue", "saturation" or "value".
+ * @private
+ */
+Blockly.FieldColourSlider.prototype.setGradient_ = function(node, channel) {
+  var gradient = this.createColourStops_(channel).join(',');
+  goog.style.setStyle(node, 'background',
+        '-moz-linear-gradient(left, ' + gradient + ')');
+  goog.style.setStyle(node, 'background',
+        '-webkit-linear-gradient(left, ' + gradient + ')');
+  goog.style.setStyle(node, 'background',
+        '-o-linear-gradient(left, ' + gradient + ')');
+  goog.style.setStyle(node, 'background',
+        '-ms-linear-gradient(left, ' + gradient + ')');
+  goog.style.setStyle(node, 'background',
+        'linear-gradient(left, ' + gradient + ')');
+};
+
+/**
+ * Update the readouts and slider backgrounds after value has changed.
+ * @private
+ */
+Blockly.FieldColourSlider.prototype.updateDom_ = function() {
+  if (this.hueSlider_) {
+    // Update the slider backgrounds
+    this.setGradient_(this.hueSlider_.getElement(), 'hue');
+    this.setGradient_(this.saturationSlider_.getElement(), 'saturation');
+    this.setGradient_(this.brightnessSlider_.getElement(), 'brightness');
+
+    // Update the readouts
+    var hsv = goog.color.hexToHsv(this.getValue());
+    this.hueReadout_.textContent = Math.floor(100 * hsv[0] / 360).toFixed(0);
+    this.saturationReadout_.textContent = Math.floor(100 * hsv[1]).toFixed(0);
+    this.brightnessReadout_.textContent = Math.floor(100 * hsv[2] / 255).toFixed(0);
+  }
+};
+
+/**
+ * Update the slider handle positions
+ */
+Blockly.FieldColourSlider.prototype.updateSliderHandles_ = function() {
+  if (this.hueSlider_) {
+    var hsv = goog.color.hexToHsv(this.getValue());
+    this.hueSlider_.animatedSetValue(hsv[0]);
+    this.saturationSlider_.animatedSetValue(hsv[1]);
+    this.brightnessSlider_.animatedSetValue(hsv[2]);
+  }
+};
+
+/**
+ * Get the text from this field.  Used when the block is collapsed.
+ * @return {string} Current text.
+ */
+Blockly.FieldColourSlider.prototype.getText = function() {
+  var colour = this.colour_;
+  // Try to use #rgb format if possible, rather than #rrggbb.
+  var m = colour.match(/^#(.)\1(.)\2(.)\3$/);
+  if (m) {
+    colour = '#' + m[1] + m[2] + m[3];
+  }
+  return colour;
+};
+
+/**
+ * Create label and readout DOM elements, returning the readout
+ * @param {string} labelText - Text for the label
+ * @return {Array} The container node and the readout node.
+ * @private
+ */
+Blockly.FieldColourSlider.prototype.createLabelDom_ = function(labelText) {
+  var labelContainer = document.createElement('div');
+  labelContainer.setAttribute('class', 'scratchColourPickerLabel');
+  var readout = document.createElement('span');
+  readout.setAttribute('class', 'scratchColourPickerReadout');
+  var label = document.createElement('span');
+  label.setAttribute('class', 'scratchColourPickerLabelText');
+  label.innerHTML = labelText;
```

Comment:
I think this can be .textContent = labelText  instead to avoid arbitrary things being dumped in via .innerHTML

Reply:
Fixed 5f8184d, looks like i missed one sorry